controllers/helmchart_controller.go

```diff
@@ -87,12 +89,14 @@ func (r *HelmChartReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// set ownership reference so chart is garbage collected on
 	// repository removal
-	r.setOwnerRef(ctx, &chart, repository)
+	if err := r.setOwnerRef(ctx, &chart, repository); err != nil {
+		log.Error(err, "failed to set owner reference")
+	}
 
 	// try to pull chart
 	pulledChart, err := r.sync(repository, *chart.DeepCopy())
 	if err != nil {
-		log.Info("Helm chart sync failed", "error", err.Error())
+		log.Error(err, "Helm chart sync failed")
```

Comment:
Do we want a stack trace here? The reason I'm using `Info` is because there is no warning func in logger

Reply:
I would expect the stack trace to be configurable using verbosity.

Reply:
But maybe I am assuming too much.

Reply:
Yes, I think the output happens because of a flag set in `main.go`: https://github.com/fluxcd/source-controller/blob/master/main.go#L73

Reply:
So to answer your question: I think we do not always want a stack trace there, but I think it is good practice to log everything as what they are, and provide the necessary flags to tune the verbosity of the logs.

Reply:
ok, we can add cmd flags for users to configure zap, dev mode can't be parsed by Fluentd or Logstash, I think the default format should be json and use dev only when running locally.